pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -824,10 +832,10 @@ func (k *KSMCheck) sendTelemetry(s sender.Sender) {
 	// reset the cache for the next check run
 	defer k.telemetry.reset()
 
-	s.Gauge(ksmMetricPrefix+"telemetry.metrics.count.total", float64(k.telemetry.getTotal()), "", k.instance.Tags)
-	s.Gauge(ksmMetricPrefix+"telemetry.unknown_metrics.count", float64(k.telemetry.getUnknown()), "", k.instance.Tags) // useful to track metrics that aren't mapped to DD metrics
+	s.Gauge(ksmMetricPrefix+"telemetry.metrics.count.total", float64(k.telemetry.getTotal()), "", k.globalTags)
+	s.Gauge(ksmMetricPrefix+"telemetry.unknown_metrics.count", float64(k.telemetry.getUnknown()), "", k.globalTags)
 	for resource, count := range k.telemetry.getResourcesCount() {
-		s.Gauge(ksmMetricPrefix+"telemetry.metrics.count", float64(count), "", append(k.instance.Tags, "resource_name:"+resource))
+		s.Gauge(ksmMetricPrefix+"telemetry.metrics.count", float64(count), "", append(k.globalTags, "resource_name:"+resource))
```

Comment:
I'm wondering it the real problem was due to the `append` in this line,

Reply:
It's both append() and a data race at the same time. `append()` won't copy the underlying array if length != capacity. 

Because we now sort k.globalTags and remove duplicates, k.globalTags won't be modified and every `append()` call will copy it.